src/screens/ChooseWallet.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { t } from 'ttag';
+
+import wallet from '../utils/wallet';
+import helpers from '../utils/helpers';
+import hathorLib from '@hathor/wallet-lib';
+import logo from '../assets/images/hathor-logo.png';
+import InitialImages from '../components/InitialImages';
+import ModalResetAllData from '../components/ModalResetAllData';
+import colors from '../index.scss';
+import ReactLoading from 'react-loading';
+import $ from 'jquery';
+
+/**
+ * Show list of wallets for user to choose from and ask for PIN.
+ *
+ * @memberof Screens
+ */
+class ChooseWallet extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // update storage in case it did not support multiple wallets (wallet upgrade)
+    const wallets = hathorLib.storage.store.getListOfWallets();
+    if (!wallets || Object.keys(wallets).length === 0){
```

Comment:
If the user removes all his wallets, are we going to assume that it's a storage upgrade?

Reply:
If the user removes all his wallets, he will go to the initial screens, where he will add a new one, so the list won't be empty. The only way we get to this condition on the ChooseWallet screen is during a wallet upgrade.